src/components/tests/tour.test.tsx

```diff
@@ -0,0 +1,223 @@
+import { createMount } from '@material-ui/core/test-utils';
+import React from 'react';
+import DashboardTour from '../tour';
+import { SpotlightValue } from '../spotlight-value';
+import { ReactWrapper } from 'enzyme';
+import { OmniPanelViewIndex } from '../dashboard';
+
+/*react-leaflet does not work well on jsdom,
+suppress warnings of not being able to find elements from leaflet*/
+jest.spyOn(console, 'warn').mockImplementation(() => {});
+
+const mount = createMount();
+
+describe('Dashboard Tour', () => {
+  let tourState: boolean = true;
+  let setTourState: jest.Mock = jest.fn();
+  let setTourShowOmniPanel: jest.Mock = jest.fn();
+  let setTourSettingsAndOmniPanel: jest.Mock = jest.fn();
+  let doorSpotlight: SpotlightValue<string> = { value: 'main_door' };
+  let setDoorSpotlight: jest.Mock = jest.fn();
+
+  let root: ReactWrapper;
+  let expectedText: string;
+  let stepText: string;
+
+  beforeEach(() => {
+    const tourProps = {
+      tourState,
+      setTourState,
+      setTourShowOmniPanel,
+      setTourSettingsAndOmniPanel,
+      OmniPanelViewIndex,
+      doorSpotlight,
+      setDoorSpotlight,
+    };
+    root = mount(<DashboardTour tourProps={tourProps} />);
+  });
+
+  afterEach(() => {
+    root?.unmount();
+  });
+
+  const wait = (delay = 0) => new Promise((resolve) => setTimeout(resolve, delay));
+
+  it('renders without crashing', () => {
+    expect(root.html()).toMatchSnapshot();
```

Comment:
Same as the comments of the snapshots before +

root.html() it's converting the `HTML` to string. You can remove the `.html()` and it'll generate a Json format. Give it a try.

Ditto for all snapshots in this file

Reply:
Oh okay I'll change that

Reply:
Using the JSON format causes the snapshot file to be really long though, because I can't use a shallow render and perform the steps of finding the elements to click etc. there will be a lot of material ui html generated too
const root = shallow(<DashboardTour.../>) will return the Tour component only without any child elements in the render tree
this is what a partial snapshot of the shallow render looks like
![shallow render tour](https://user-images.githubusercontent.com/28668944/94135672-6027ec00-fe96-11ea-8e6b-48fdc8319cd1.png)

Reply:
Mmm.. one last try, can you check if there is a way to make something like this?  `expect(root.find('.reactour__helper')).toMatchSnapshot();`

To snap just a smaller part